unittests/firmware/main.c

```diff
@@ -29,7 +31,12 @@ int main() {
         if ((rcv == ' ') || (rcv == '\t') || (rcv == '\r') || (rcv == '\n')) {
             // whitespace
             if (idx > 0) {
-                buf[idx++] = '\0';
+                // pad rest of the string with null characters
```

Comment:
I was seeing a weird issue on ZCU106, where somehow memory was getting corrupted unless I padded the end of the string with all null characters.  This shouldn't be necessary (only one null character at the end of the string is required), but perhaps it was an indicator of something else that was going wrong.